src/main/java/seedu/address/logic/commands/ListCommand.java

```diff
@@ -5,30 +5,155 @@
+
+        switch (action) {
+        case "":
+            model.refreshAllFilteredLists();
+            return new CommandResult(MESSAGE_SUCCESS);
+        case "unpaired":
+            model.updateFilteredElderlyList(getUnPairedElderlyPredicate(friendlyLink));
+            model.updateFilteredVolunteerList(getUnPairedVolunteerPredicate(friendlyLink));
+            model.updateFilteredPairList(showAllPairPredicate);
+            return new CommandResult(String.format(MESSAGE_SUCCESS_LIST_UNPAIRED,
+                    model.getFilteredVolunteerList().size(), model.getFilteredElderlyList().size()));
+        case "paired":
+            model.updateFilteredElderlyList(getPairedElderlyPredicate(friendlyLink));
+            model.updateFilteredVolunteerList(getPairedVolunteerPredicate(friendlyLink));
+            model.updateFilteredPairList(showAllPairPredicate);
+            return new CommandResult(String.format(MESSAGE_SUCCESS_LIST_PAIRED,
+                    model.getFilteredVolunteerList().size(), model.getFilteredElderlyList().size()));
+        default:
+            throw new RuntimeException("unreachable");
+        }
+    }
+
+    /**
+     * Returns the paired elderly predicate based on the filtered pair list.
+     *
+     * @param friendlyLink Application cache.
+     * @return Elderly predicate.
+     */
+    public Predicate<Elderly> getPairedElderlyPredicate(FriendlyLink friendlyLink) {
+        requireNonNull(friendlyLink);
+        List<Pair> pairList = friendlyLink.getPairList();
+        List<Elderly> pairedElderlyList =
+                pairList.stream().map(pair -> pair.getElderly()).collect(Collectors.toList());
+
+        return elderly -> pairedElderlyList.contains(elderly);
+    }
+
+    /**
+     * Returns the paired volunteer predicate based on the filtered pair list.
+     *
+     * @param friendlyLink Application cache.
+     * @return Volunteer predicate.
+     */
+    public Predicate<Volunteer> getPairedVolunteerPredicate(FriendlyLink friendlyLink) {
+        requireNonNull(friendlyLink);
+        List<Pair> pairList = friendlyLink.getPairList();
+        List<Volunteer> pairedVolunteerList =
+                pairList.stream().map(pair -> pair.getVolunteer()).collect(Collectors.toList());
+
+        return volunteer -> pairedVolunteerList.contains(volunteer);
+    }
+
+    /**
+     * Returns the unpaired elderly predicate based on the filtered pair list.
+     *
+     * @param friendlyLink Application cache.
+     * @return Elderly predicate.
+     */
+    public Predicate<Elderly> getUnPairedElderlyPredicate(FriendlyLink friendlyLink) {
+        requireNonNull(friendlyLink);
+        List<Pair> pairList = friendlyLink.getPairList();
+        List<Elderly> pairedElderlyList =
+                pairList.stream().map(pair -> pair.getElderly()).collect(Collectors.toList());
+
+        return elderly -> !pairedElderlyList.contains(elderly);
```

Comment:
```suggestion
        requireNonNull(friendlyLink);
        return elderly -> !(getPairedElderlyPredicate(friendlyLink)(elderly));
```
Try if this works (same for volunteer)

Reply:
It doesn't work. Thanks for trying anyways